_pytest/logging.py

```diff
@@ -2,15 +2,73 @@
 
 import logging
 from contextlib import closing, contextmanager
+import re
 import six
 
 import pytest
 import py
 
 
+LOGLEVEL_COLOROPTS = {
+    logging.CRITICAL: {'red'},
+    logging.ERROR: {'red', 'bold'},
+    logging.WARNING: {'yellow'},
+    logging.WARN: {'yellow'},
+    logging.INFO: {'green'},
+    logging.DEBUG: {'purple'},
+    logging.NOTSET: set(),
+}
+
 DEFAULT_LOG_FORMAT = '%(filename)-25s %(lineno)4d %(levelname)-8s %(message)s'
 DEFAULT_LOG_DATE_FORMAT = '%H:%M:%S'
 
+# move to __init__ of ColoredLevelFormatter?
+LEVELNAME_FMT_REGEX = re.compile(r'%\(levelname\)([+-]?\d*s)')
+
+
+class ColoredLevelFormatter(logging.Formatter):
```

Comment:
To test this, probably the way to go is to unittest the `ColoredLevelFormatter` class directly: create an instance and pass different `LogRecord` instances and ensure they have the correct markup applied. Also interesting to use custom log levels and ensure they don't have any markup.

Reply:
Unfortunately this does not work, since `tw.markup()` does not add ANSI escape sequences around the string in the unit tests:

```
(Pdb++) tw.markup('abc', red=True)
'abc'
```

Any ideas?

Reply:
I'll do some mocking and then it should work.

Reply:
ping @nicoddemus